### ISSUE TITLE
Cookies policy and consent

### ISSUE THREAD

**Issue by cathydutton** [DEFRA/design-system-backlog#9]
## What
Help users manage their personal data by telling them when you store cookies on their device.

## Why
We must have users consent to store cookies or similar technologies on their device

## How

* Ask users for consent
* Provide information on cookies used
* Allow users to set preferences
* Allow users to change preferences
* Inform users of any changes

**Comment by cathydutton** [comment]
![image](https://user-images.githubusercontent.com/5614324/75340298-2522f300-588a-11ea-907b-8e7a156d873b.png)

**Comment by matthewsolle** [comment]
Reference

https://www.gov.uk/service-manual/technology/working-with-cookies-and-similar-technologies

https://docs.publishing.service.gov.uk/manual/cookie-consent-on-govuk.html

https://github.com/alphagov/govuk-design-system-backlog/issues/12

https://github.com/alphagov/govuk-design-system-backlog/issues/13

**Comment by tomf87** [comment]
Just so everyone's aware, on the design system call I took part in I heard that GDS have no plans to work on a pattern for cookie banners until after they've looked at a technical solution to share cookies across departmental sites, which could take up to two years from now.

**Comment by tomf87** [comment]
Pretty sure pattern 6 would be non-compliant under a stricter interpretation of GDPR, as the button design leads the user towards accepting cookies.

**Comment by cathydutton** [comment]
> Pretty sure pattern 6 would be non-compliant under a stricter interpretation of GDPR, as the button design leads the user towards accepting cookies.

Thats the one Gov.uk blogs are now using

**Comment by cathydutton** [comment]
This is from the DWP backlog, I think its a good way of communicating the different flows and messages. 

The part I'm unsure about is how users update their preferences. Is it enough to have that option buried in the cookies page?

![75032943-a866d200-54a1-11ea-9c00-24195eefdf42](https://user-images.githubusercontent.com/5614324/75342129-b5af0280-588d-11ea-9bc7-81ee1152c0fa.png)

**Comment by tomf87** [comment]
> > Pretty sure pattern 6 would be non-compliant under a stricter interpretation of GDPR, as the button design leads the user towards accepting cookies.
> 
> Thats the one Gov.uk blogs are now using

Yeah I think it depends how you interpret the policy, I'm just mentioning it because where I worked before we decided to avoid using different button styles as we wanted to avoid introducing any bias.

**Comment by cathydutton** [comment]
> > > Pretty sure pattern 6 would be non-compliant under a stricter interpretation of GDPR, as the button design leads the user towards accepting cookies.
> > 
> > 
> > Thats the one Gov.uk blogs are now using
> 
> Yeah I think it depends how you interpret the policy, I'm just mentioning it because where I worked before we decided to avoid using different button styles as we wanted to avoid introducing any bias.

Good point, theres a thread on dark patterns around consent here - https://twitter.com/yahoo_pete/status/1230562192144994307.

I wonder if we need 3 buttons Accept all, Reject all and manage preferences?

**Comment by jOnoNe** [comment]
With the absence of cookie sharing across urls, is there anything we can do to reduce the need for users clicking multiple banners in the same cross-domain journey?

**Comment by cathydutton** [comment]
Collated examples of current designs from across Gov

![cookie-consent](https://user-images.githubusercontent.com/5614324/75565853-9effdb80-5a46-11ea-962e-a8479a3d6bc4.png)

**Comment by peter-jordan** [comment]
Going back to the comments on the GOV.UK blogs design. A quick and crude comparison of the 'before' and 'after' volume of sessions shows that www.gov.uk is getting around twice as much acceptance of cookies compared to the blogging platform.

So you could argue **words** as much as **button styles**.

**Comment by peter-jordan** [comment]
@cathydutton What URL got you the 'Can I get Legal Aid?'/ Notify banner? I can't replicate it.

**Comment by cathydutton** [comment]
> With the absence of cookie sharing across urls, is there anything we can do to reduce the need for users clicking multiple banners in the same cross-domain journey?

Is this happening with services using IDM? May be worth asking wider Gov teams to see if anyone else has dealt with this.

**Comment by cathydutton** [comment]
> @cathydutton What URL got you the 'Can I get Legal Aid?'/ Notify banner? I can't replicate it.

I got it from a Slack thread so not sure how you get to it.

**Comment by cathydutton** [comment]
@jOnoNe - planned work to look into this at GDS - https://www.gov.uk/service-manual/technology/working-with-cookies-and-similar-technologies#discovery-into-remembering-consent-across-domains-and-subdomains

**Comment by cathydutton** [comment]
<img width="1236" alt="Screen Shot 2020-03-18 at 12 13 17" src="https://user-images.githubusercontent.com/5614324/76959631-e1b02780-6911-11ea-9da5-b65563993247.png">

DVSA example - using two buttons instead of a link for settings

**Comment by cathydutton** [comment]
[flow.pdf](https://github.com/DEFRA/design-discussions/files/4354039/flow.pdf)

**Comment by cathydutton** [comment]
![flow](https://user-images.githubusercontent.com/5614324/77059997-c73c8380-69cf-11ea-9f15-3b20cd8b2524.jpg)

**Comment by MalcolmVonMoJ** [comment]
In _**Check if you can get Legal Aid**_, we were using pattern number 3 until recently.  

We have now changed to this pattern as an interim whilst research is ongoing, so it might not be our final decision.  
![image](https://user-images.githubusercontent.com/32877315/77152910-bf90e380-6a90-11ea-897d-61b74ed1b2d0.png)

Our reasoning for choosing this was:
- We wanted something that was visibly different from the main GOV.UK cookie banner, so users would recognise that this is not the one they might've just accepted/rejected.  
- We wanted something that was more obvious than the then-current banner (number 3 in my image at the top of this thread), so users would be less inclined to just use the service without ignoring it.  
- Counter to that, we also didn't want to be too intrusive.
- We didn't want a simple _**yes**_/_**no**_ option as we have mandatory cookies for Welsh language and so to say _**no**_ was misleading and to say _**accept all**_ and _**accept mandatory**_ is starting to get a tad wordy.

So, we kept the current choices (_**yes**_/_**settings**_) and only amended the look and feel of the banner for the now.  Our brilliant user researcher and exquisite content designer have not put this to bed yet and are continuing to look at all the above points.  

However, the moment we made this change (March 6th, 2 weeks ago today), we noticed a significant increase in users accepting cookies.  Approximately double.  This still only puts us at 40% of pre-GDPR levels in round figures.  

Google Analytics details:
![image](https://user-images.githubusercontent.com/32877315/77154127-0253bb00-6a93-11ea-91f0-ec9ad4609f95.png)

**Comment by cathydutton** [comment]
<img width="696" alt="Screen Shot 2020-03-25 at 12 45 06" src="https://user-images.githubusercontent.com/5614324/77537737-7d9fdd00-6e96-11ea-8e04-00d786a057d6.png">

Latest update from Gov.uk

**Comment by cathydutton** [comment]
[Artboard.pdf](https://github.com/DEFRA/design-discussions/files/4402784/Artboard.pdf)

**Comment by cathydutton** [comment]
![Artboard](https://user-images.githubusercontent.com/5614324/77913848-e5866700-728c-11ea-8a2e-df56461bee95.jpg)
